### PR TITLE
Selectively enable page scrolling in the gallery animation demo

### LINKED ISSUE #9603: Animation Demo should not show sliders when swiping horizontally
Feedback from Material Design Review of the Gallery.  The feedback was marked "P1" so would block future releases of the Gallery:

"Swiping horizontally on the animation section shows the sliders, which it shouldn't"

### PR DESCRIPTION
**Pull request by HansMuller**
Only enable page scrolling when the animation demo has been scrolled up to its "row" layout.

Render the selection indicators correctly in "column" layout - only the selected section indicator should be highlighted.

Fixed a ScrollPhysics problem: the applyTo() method implementation[s] didn't work correctly when the ScrollPhysics had a non-null parent.

Fixes #9603

### CODE REVIEW

File: packages/flutter/lib/src/widgets/scrollable.dart [hunk 1]
```diff
@@ -256,8 +256,16 @@ class ScrollableState extends State<Scrollable> with TickerProviderStateMixin
...
-    return widget.physics?.runtimeType != oldWidget.physics?.runtimeType
-        || widget.controller?.runtimeType != oldWidget.controller?.runtimeType;
+    ScrollPhysics newPhysics = widget.physics;
+    ScrollPhysics oldPhysics = oldWidget.physics;
+    do {
+      if (newPhysics?.runtimeType != oldPhysics?.runtimeType)
+        return true;
+      newPhysics = newPhysics?.parent;
+      oldPhysics = oldPhysics?.parent;
+    } while(newPhysics != null || oldPhysics != null);
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/scrollable.dart]: nit: space after while

**Comment by Hixie** [comment]
Is there some way we can factor out the `parent?.applyTo(ancestor) ?? ancestor` incantation?

Also, is there some way we can explicitly check that all the Physics classes we have invented handle being applied to a chain and handle being applied as part of a chain (at the top, middle, and end of the chain)? Because that would be a really valuable test to have...

**Comment by HansMuller** [comment]
I've factored `parent?.applyTo(ancestor) ?? ancestor` into a protected method and added  tests for the ScrollPhysics subcasses.

File: packages/flutter/test/widgets/scroll_physics_test.dart [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+    expect(types(never.applyTo(always.applyTo(page.applyTo(bounce.applyTo(clamp))))),
+      'NeverScrollableScrollPhysics AlwaysScrollableScrollPhysics PageScrollPhysics BouncingScrollPhysics ClampingScrollPhysics');
+
+    expect(types(always.applyTo(page.applyTo(bounce.applyTo(clamp.applyTo(never))))),
+      'AlwaysScrollableScrollPhysics PageScrollPhysics BouncingScrollPhysics ClampingScrollPhysics NeverScrollableScrollPhysics');
+
+    expect(types(page.applyTo(bounce.applyTo(clamp.applyTo(never.applyTo(always))))),
+      'PageScrollPhysics BouncingScrollPhysics ClampingScrollPhysics NeverScrollableScrollPhysics AlwaysScrollableScrollPhysics');
+  });
```
Comment (Hixie) [inline packages/flutter/test/widgets/scroll_physics_test.dart]: you should probably also test toString on a couple of these, to get some free extra coverage

**Comment by Hixie** [comment]
LGTM

**Pull request merged**